tests/test_event_detection.py

```diff
@@ -1,7 +1,180 @@
+    mat_all_evs = matlab_file['event_times'][:, 0]
+
+    # Get python events by running find_events on demo data
+    # <----------SIMPLIFY BY GETTING PRE-SAVED PICKLE FILES FOR THIS INSTEAD----------->
+    subj_ids = [str(id) for id in range(1, 10 + 1)]  # subject IDs 1-10
+
+    # set parameters
+    samp_freq = 600
+    n_times = 600
+    # fequency values (Hz) over which to calculate TFR
+    freqs = list(range(1, 60 + 1))
+    times = np.arange(n_times) / samp_freq  # seconds
+    event_band = [15, 29]  # beta band (Hz)
+    thresh_fom = 6.0  # factor-of-the-median threshold
+
+    # get python evs
+    py_all_evs = list()
+    for _, id_val in enumerate(subj_ids):
+        fname = op.join(
+            data_dir, 'data','prestim_humandetection_600hzMEG_subject' + id_val + '.mat')
+        raw_data = loadmat(fname)
+        data = raw_data['prestim_raw_yes_no']  # MEG time series (trials x samples)
+        # calculate TFR
+        tfrs = se.tfr(data, freqs, samp_freq)
+
+        # find events
+        subj_evs = se.find_events(tfr=tfrs, times=times, freqs=freqs, event_band=event_band,
+                                  thresholds=None, threshold_FOM=thresh_fom, find_method=method)
+        py_all_evs.append(subj_evs)
+
+    # put from cell array/dict into sets
+    py_evs = np.array([])
+    for subj_idx, id_val in enumerate(subj_ids):
+        py_evs = np.append(py_evs,np.array([len(trial) for trial in py_all_evs[subj_idx]]))
+    mat_evs = np.array([trial.shape[1] for trial in mat_all_evs])
+
+    # Overall check that same # of evs detected (or we adapt this to a tolerable # of difference)
+    assert np.array_equal(
+        mat_evs, py_evs), "Should be same number of events across all trials"
+
+    # Once assured same num of evs, extract event characteristics into np arrays
+    # matlab evs
+    mat_times = np.array([])
+    mat_max_freq = np.array([])
+    mat_low_freq = np.array([])
+    mat_high_freq = np.array([])
+    mat_onsets = np.array([])
+    mat_offsets = np.array([])
+    mat_power = np.array([])
+    trial_ind = -1
+    for trial in mat_all_evs:
+        trial_ind += 1
+        mat_times = [np.append(mat_times, event) for event in trial]
+        mat_max_freq = [np.append(mat_max_freq, event)
+                      for event in matlab_file['event_maxfreqs'][trial_ind]]
+        mat_low_freq = [np.append(mat_low_freq, event)
+                      for event in matlab_file['event_lowfreqs'][trial_ind]]
+        mat_high_freq = [np.append(mat_high_freq, event)
+                       for event in matlab_file['event_highfreqs'][trial_ind]]
+        mat_onsets = [np.append(mat_onsets, event)
+                     for event in matlab_file['event_onsets'][trial_ind]]
+        mat_offsets = [np.append(mat_offsets, event)
+                      for event in matlab_file['event_offsets'][trial_ind]]
+        mat_power = [np.append(mat_power, event)
+                    for event in matlab_file['event_powers'][trial_ind]]
```

Comment:
I think these list comprehensions can probably be put into a single for loop that would be more efficient and more readable.

Reply:
Agreed, that's smoother. Will reconfigure.